src/lib/utils/bitvector.h

```diff
@@ -0,0 +1,1327 @@
+       */
+      template <bitvectorish OutT = bitvector_base<AllocatorT>>
+      auto subvector(size_type pos, std::optional<size_type> length = std::nullopt) const {
+         size_type bitlen = length.value_or(size() - pos);
+         BOTAN_ARG_CHECK(pos + bitlen <= size(), "Not enough bits to copy");
+
+         OutT newvector(bitlen);
+
+         // Handle bitvectors that are wrapped in strong types
+         auto& newvector_unwrapped = detail::unwrap(newvector);
+         using unwrapped_type = std::remove_cvref_t<decltype(newvector_unwrapped)>;
+         static_assert(is_bitvector_v<unwrapped_type>);
+
+         if(bitlen > 0) {
+            if(pos % 8 == 0) {
+               copy_mem(
+                  newvector_unwrapped.m_blocks,
+                  std::span{m_blocks}.subspan(block_index(pos), block_index(pos + bitlen - 1) - block_index(pos) + 1));
+            } else {
+               BitRangeOperator<const bitvector_base<AllocatorT>, BitRangeAlignment::no_alignment> from_op(
+                  *this, pos, bitlen);
+               BitRangeOperator<unwrapped_type> to_op(detail::unwrap(newvector_unwrapped), 0, bitlen);
+               range_operation([](auto /* to */, auto from) { return from; }, to_op, from_op);
+            }
+
+            newvector_unwrapped.zero_unused_bits();
+         }
+
+         return newvector;
+      }
+
+      /// @}
+
+      /// @name Operators
+      ///
+      /// @{
+
+      auto operator~() {
+         auto newbv = *this;
+         newbv.flip();
+         return newbv;
+      }
+
+      template <bitvectorish OtherT>
+      auto& operator|=(const OtherT& other) {
+         full_range_operation([]<std::unsigned_integral BlockT>(BlockT lhs, BlockT rhs) -> BlockT { return lhs | rhs; },
+                              *this,
+                              detail::unwrap(other));
+         return *this;
+      }
+
+      template <bitvectorish OtherT>
+      auto& operator&=(const OtherT& other) {
+         full_range_operation([]<std::unsigned_integral BlockT>(BlockT lhs, BlockT rhs) -> BlockT { return lhs & rhs; },
+                              *this,
+                              detail::unwrap(other));
+         return *this;
+      }
+
+      template <bitvectorish OtherT>
+      auto& operator^=(const OtherT& other) {
+         full_range_operation([]<std::unsigned_integral BlockT>(BlockT lhs, BlockT rhs) -> BlockT { return lhs ^ rhs; },
+                              *this,
+                              detail::unwrap(other));
+         return *this;
+      }
+
+      /// @}
+
+      /// @name Constant Time Operations
+      ///
+      /// @{
+
+      /**
+       * Implements::
+       *
+       *    if(condition) {
+       *       *this ^= other;
+       *    }
+       *
+       * omitting runtime dependence on any of the parameters.
+       */
+      template <bitvectorish OtherT>
+      void ct_conditional_xor(bool condition, const OtherT& other) {
```

Comment:
Here again `CT::Choice` may be a good option. In general for constant time code I'd suggest avoiding `bool` - I have in the past seen specific cases where LLVM realizes that `bool` can have just two possible values and that's it's reasonable for fork the execution between the possible paths using a conditional jump.